### PR TITLE
PTX-22802 [Longevity]Add Longevity Triggers for V-cluster

### PR DESCRIPTION
**Pull request by spai-px**
**What this PR does / why we need it**:
This PR adds longevity test triggers. 

**Which issue(s) this PR fixes** (optional)
Closes # PTX-22802

**Special notes for your reviewer**:
Will post results once testing is done.

### CODE REVIEW

File: tests/testTriggers.go [hunk 1]
```diff
@@ -9933,6 +9947,403 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
...
+	var appNS string
+	fioOptions := vcluster.FIOOptions{
+		Name:      "mytest",
+		IOEngine:  "libaio",
+		RW:        "randwrite",
+		BS:        "4k",
+		NumJobs:   1,
+		Size:      "500m",
+		TimeBased: true,
+		Runtime:   "600s",
```
Comment (dbhatnagar-px) [inline tests/testTriggers.go:9982]: Keep the runtime as configurable - maybe reading from configmap. Keep default as 1 day,

File: tests/testTriggers.go [hunk 2]
```diff
@@ -9933,6 +9947,403 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
...
+		// Create PVC on VCluster
+		appNS = scName + "-ns"
+		pvcName, err = vc.CreatePVC("", scName, appNS, "")
+		if err != nil {
+			UpdateOutcome(event, err)
+		}
+
+		jobName := "fio-job"
+		// Create FIO Deployment on VCluster using the above PVC
+		err = vc.CreateFIODeployment(pvcName, appNS, fioOptions, jobName)
```
Comment (dbhatnagar-px) [inline tests/testTriggers.go:10026]: Increase timeout param in WaitForFIOCompletion

File: tests/testTriggers.go [hunk 3]
```diff
@@ -9933,6 +9947,403 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
...
+	}
+
+	err = vc.CreateAndWaitVCluster()
+	if err != nil {
+		UpdateOutcome(event, err)
+		return
+	}
+
+	envValueIterations := vcluster.ReadEnvVariable("VCLUSTER_TOTAL_ITERATIONS")
+	envValueBatch := vcluster.ReadEnvVariable("VCLUSTER_PARALLEL_APPS")
```
Comment (dbhatnagar-px) [inline tests/testTriggers.go:10082]: Please keep these in configmap

File: tests/testTriggers.go [hunk 4]
```diff
@@ -9933,6 +9947,403 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
...
+	}
+	fioOptions := vcluster.FIOOptions{
+		Name:      "mytest",
+		IOEngine:  "libaio",
+		RW:        "randwrite",
+		BS:        "4k",
+		NumJobs:   1,
+		Size:      "500m",
+		TimeBased: true,
+		Runtime:   "100s",
```
Comment (dbhatnagar-px) [inline tests/testTriggers.go:10111]: Increase the time here as well

**Review by dbhatnagar-px — CHANGES_REQUESTED** [review]

File: tests/testTriggers.go (hunk 1 above)
Comment (spai-px) [reply tests/testTriggers.go]: Done. defined the variable to take default as 86,400 seconds and have configured a function to read the data if present in config map.

File: tests/testTriggers.go (hunk 2 above)
Comment (spai-px) [reply tests/testTriggers.go]: Done

File: tests/testTriggers.go (hunk 3 above)
Comment (spai-px) [reply tests/testTriggers.go]: Done. picking it from config map.

File: tests/testTriggers.go (hunk 4 above)
Comment (spai-px) [reply tests/testTriggers.go]: Done. picking this from config-map

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

File: tests/longevity/longevity_helper.go [hunk 5]
```diff
@@ -1361,6 +1380,39 @@ func populateIntervals() {
 	triggerInterval[ReallocateSharedMount][2] = 24 * baseInterval
 	triggerInterval[ReallocateSharedMount][1] = 27 * baseInterval
 
+	triggerInterval[CreateAndRunFioOnVcluster][10] = 1 * baseInterval
+	triggerInterval[CreateAndRunFioOnVcluster][9] = 3 * baseInterval
+	triggerInterval[CreateAndRunFioOnVcluster][8] = 6 * baseInterval
+	triggerInterval[CreateAndRunFioOnVcluster][7] = 9 * baseInterval
+	triggerInterval[CreateAndRunFioOnVcluster][6] = 12 * baseInterval
+	triggerInterval[CreateAndRunFioOnVcluster][5] = 15 * baseInterval // Default global chaos level, 3 hrs
```
Comment (kphalgun-px) [inline tests/longevity/longevity_helper.go:1388]: 👍

File: tests/testTriggers.go [hunk 6]
```diff
@@ -9933,6 +9962,425 @@ func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan
...
+	reclaimPolicyDelete := v1.PersistentVolumeReclaimDelete
+	bindMode := storageApi.VolumeBindingImmediate
+	scObj := storageApi.StorageClass{
+		ObjectMeta:        v1obj,
+		Provisioner:       k8s.CsiProvisioner,
+		Parameters:        params,
+		ReclaimPolicy:     &reclaimPolicyDelete,
+		VolumeBindingMode: &bindMode,
+	}
+	// Applying each extra option to Storage class definition
```
Comment (kphalgun-px) [inline tests/testTriggers.go:10373]: Good to see functional options pattern.

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**